zipkin-lens/src/components/TracePage/TraceSummary.jsx

```diff
@@ -94,15 +102,28 @@ const TraceSummary = React.memo(({ traceSummary }) => {
     return rerootedTree.filter(span => !allHiddenSpanIds[span.spanId]);
   }, [rerootedTree, childrenHiddenSpanIds]);
 
-  // Find the minumum and maximum timestamps in the shown spans.
-  const startTs = useMemo(() => minBy(rerootedTree, 'timestamp').timestamp, [rerootedTree]);
-  const endTs = useMemo(() => rerootedTree.map((span) => {
-    let ts = span.timestamp;
-    if (span.duration) {
-      ts += span.duration;
-    }
-    return ts;
-  }).reduce((a, b) => Math.max(a, b)), [rerootedTree]);
+  // Find the minumum and maximum timestamps in shown spans.
+  const { startTs, endTs } = useMemo(() => {
```

Comment:
It would be nice if we didn't have to special-case for 0 timestamp any time we use a span in the UI. Do you think it could make sense to add a `normalizedTimestamp` to span in trace-cleaner.js that is set to the trace timestamp when timestamp isn't present? We could use that everywhere except in the detailed view of the span where we should indicate the timestamp wasn't present.

Reply:
I agree with you.

Reply:
this will assume trace has a timestamp :D